src/plugins/hapi.js

```diff
@@ -56,93 +57,64 @@ function createWrapDispatch (tracer, config) {
-    file: 'lib/request.js',
-    patch (Generator, tracer, config) {
-      this.wrap(Generator.prototype, 'request', createWrapGenerate(tracer, config))
-    },
-    unpatch (Generator) {
-      this.unwrap(Generator.prototype, 'request')
-    }
-  },
-  {
-    name: 'hapi',
-    versions: ['2 - 8.4'],
-    file: 'lib/request.js',
-    patch (Request, tracer, config) {
-      this.wrap(Request.prototype, '_execute', createWrapExecute(tracer, config))
-    },
-    unpatch (Request) {
-      this.unwrap(Request.prototype, '_execute')
-    }
-  },
-  {
-    name: 'hapi',
-    versions: ['7.2 - 16'],
-    file: 'lib/connection.js',
-    patch (Connection, tracer, config) {
-      this.wrap(Connection.prototype, '_dispatch', createWrapDispatch(tracer, config))
-    },
-    unpatch (Connection) {
-      this.unwrap(Connection.prototype, '_dispatch')
-    }
-  },
-  {
-    name: '@hapi/hapi',
-    versions: ['>=17.9'],
-    file: 'lib/core.js',
-    patch (Core, tracer, config) {
-      this.wrap(Core.prototype, '_dispatch', createWrapDispatch(tracer, config))
-    },
-    unpatch (Core) {
-      this.unwrap(Core.prototype, '_dispatch')
-    }
-  },
-  {
-    name: 'hapi',
-    versions: ['>=17'],
-    file: 'lib/core.js',
-    patch (Core, tracer, config) {
-      this.wrap(Core.prototype, '_dispatch', createWrapDispatch(tracer, config))
-    },
-    unpatch (Core) {
-      this.unwrap(Core.prototype, '_dispatch')
-    }
-  },
-  {
-    name: 'hapi',
-    versions: ['2 - 7.1'],
-    file: 'lib/server.js',
-    patch (Server, tracer, config) {
-      this.wrap(Server.prototype, '_dispatch', createWrapDispatch(tracer, config))
+  }
+}
+
+function patch (name, versions, file, property, method) {
+  return {
+    name: name,
+    versions: versions,
+    file: file,
+    patch (Obj, tracer, config) {
+      this.wrap(Obj.prototype, property, method(tracer, config))
     },
-    unpatch (Server) {
-      this.unwrap(Server.prototype, '_dispatch')
+    unpatch (Obj) {
+      this.unwrap(Obj.prototype, property)
     }
   }
+}
+
+function patchDispatch (name, versions, file) {
+  return patch(name, versions, file, '_dispatch', createWrapDispatch)
+}
+
+function patchGenerate (name, versions, file, property) {
+  return patch(name, versions, file, property, createWrapGenerate)
+}
+
+function patchExecute (name, versions, file) {
+  return patch(name, versions, file, '_execute', createWrapExecute)
+}
+
+let patches = [
+  patchRequestGenerate('@hapi/hapi', ['>=17.9 <19.0']),
+  patchRequestGenerate('hapi', ['>=17.1 <19.0']),
+  patchGenerate('hapi', ['8.5 - 17.0'], 'lib/request.js', 'request'),
+  patchExecute('hapi', ['2 - 8.4'], 'lib/request.js'),
+  patchDispatch('hapi', ['7.2 - 16'], 'lib/connection.js'),
+  patchDispatch('@hapi/hapi', ['>=17.9 <19.0'], 'lib/core.js'),
+  patchDispatch('hapi', ['>=17 <19.0'], 'lib/core.js'),
+  patchDispatch('hapi', ['2 - 7.1'], 'lib/server.js')
 ]
+
+if (semver.gte(process.version, '12.0.0')) {
```

Comment:
Hapi `19.0` adds deps that only work on node>=12 so activating those patches breaks CI jobs that use older node version.